packages/nextjs/src/app-router/client/useInternalNavFun.ts

```diff
@@ -6,31 +6,48 @@ import type { NextClerkProviderProps } from '../../types';
 
 declare global {
   interface Window {
-    __clerk_internal_navFun: NonNullable<
-      NextClerkProviderProps['routerPush'] | NextClerkProviderProps['routerReplace']
+    __clerk_internal_navigations: Record<
+      string,
+      {
+        fun: NonNullable<NextClerkProviderProps['routerPush'] | NextClerkProviderProps['routerReplace']>;
+        promisesBuffer: Array<() => void> | undefined;
+      }
     >;
-    __clerk_internal_navPromisesBuffer: Array<() => void> | undefined;
   }
 }
 
+const registerNavigationType = (name: string) => {
+  if (!window.__clerk_internal_navigations) {
+    window.__clerk_internal_navigations = {};
+  }
+
+  if (!(name in window.__clerk_internal_navigations)) {
+    // @ts-ignore
+    window.__clerk_internal_navigations[name] = {};
+  }
+
+  return window.__clerk_internal_navigations[name];
+};
+
 export const useInternalNavFun = (props: {
   windowNav: typeof window.history.pushState | typeof window.history.replaceState | undefined;
   routerNav: AppRouterInstance['push'] | AppRouterInstance['replace'];
+  name: string;
 }) => {
-  const { windowNav, routerNav } = props;
+  const { windowNav, routerNav, name } = props;
   const pathname = usePathname();
   const [isPending, startTransition] = useTransition();
 
   if (windowNav) {
-    window.__clerk_internal_navFun = (to, opts) => {
+    registerNavigationType(name).fun = (to, opts) => {
       return new Promise<void>(res => {
-        if (!window.__clerk_internal_navPromisesBuffer) {
+        if (!registerNavigationType(name).promisesBuffer) {
           // We need to use window to store the reference to the buffer,
           // as ClerkProvider might be unmounted and remounted during navigations
           // If we use a ref, it will be reset when ClerkProvider is unmounted
-          window.__clerk_internal_navPromisesBuffer = [];
+          registerNavigationType(name).promisesBuffer = [];
```

Comment:
Could be simplified with:
```suggestion
          registerNavigationType(name).promisesBuffer ??= [];
```

Can remove the wrapped if that way.